client.ts

```diff
@@ -830,4 +830,36 @@ export class Client {
       versionId: response.headers.get("x-amz-version-id") || null,
     };
   }
+
+  public async bucketExists(bucketName: string): Promise<boolean> {
+    try {
+      const objects = this.listObjects({ bucketName });
+      // We don't need to fully list the objects, just check if we can start listing
+      await objects.next();
+      return true;
+    } catch (err: unknown) {
+      if (err instanceof errors.ServerError && err.statusCode === 404) {
+        return false;
+      }
+      throw err;
+    }
+  }
+
+  public async makeBucket(bucketName: string): Promise<void> {
+    await this.makeRequest({
+      method: "PUT",
+      bucketName: this.getBucketName({ bucketName }),
```

Comment:
Since we're calling `this.getBucketName`, we could specify that the `bucketName: string` parameter is optional, and if it's undefined it would fall back to `this.defaultBucket`. But I guess that even if we're not using that functionality here, `this.getBucketName()` is still serving a useful purpose of validating the bucket name.

Reply:
At first, I thought about allowing the `bucketName` to be optional, but I ultimately concluded that emphasizing clarity by requiring it explicitly was the best approach